### PR TITLE
remove resizer from dom when not in use

### PR DESCRIPTION
**Pull request by snowystinger**
Closes 

found in testing

## ✅ Pull Request Checklist:

- [ ] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [ ] Filled out test instructions.
- [ ] Updated documentation (if it already exists for this component).
- [ ] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

### CODE REVIEW

File: packages/@react-spectrum/table/src/TableView.tsx [hunk 1]
```diff
@@ -506,9 +506,9 @@ function TableVirtualizer({layout, collection, lastResizeInteractionModality, fo
           onScrollEnd={state.endScrolling}
           onScroll={onScroll}>
           {state.visibleViews[1]}
-          <div
+          {columnState.currentlyResizingColumn && <div
```
Comment (devongovett) [inline packages/@react-spectrum/table/src/TableView.tsx:509]: If this is a key, you'll need to check if it's null and not only falsy

Comment (snowystinger) [reply packages/@react-spectrum/table/src/TableView.tsx:509]: haha, it got me!

**Review by LFDanLu — APPROVED** [review]
LGTM, confirmed the resizer isn't in the DOM when resizing isn't happening. Also confirmed that falsy key columns still render the resizer when being resized

**Review by reidbarber — APPROVED** [review]
LGTM

**Pull request merged**